### ISSUE TITLE
Processing dbms.jvm.additional broken since 4.4.9

### ISSUE THREAD

**Issue by nioertel** [neo4j/docker-neo4j#363]
## Bug report

When upgrading to Neo4j 4.4.9 we realized that processing of the property `dbms.jvm.additional` is not working anymore for JVM parameters that contain spaces.

Extract from neo4j.conf:
`dbms.jvm.additional="-XX:+HeapDumpOnOutOfMemoryError" "-XX:HeapDumpPath=/tmp/" "-XX:OnOutOfMemoryError=/usr/bin/env handle-oom.sh %p /tmp/java_pid%p.hprof" "-Dlog4j.configurationFile=classpath:log4j2.json"`

Output of `cat /proc/$PID/cmdline | tr '\000' '\n'`
```
/opt/java/openjdk/bin/java
-cp
/var/lib/neo4j/plugins/*:/var/lib/neo4j/conf/*:/var/lib/neo4j/lib/*
-Xms4194304k
-Xmx4194304k
-XX:+HeapDumpOnOutOfMemoryError
-XX:HeapDumpPath=/tmp/
"-XX:OnOutOfMemoryError=/usr/bin/env
handle-oom.sh
%p
/tmp/java_pid%p.hprof"
-Dlog4j.configurationFile=classpath:log4j2.json
-Dfile.encoding=UTF-8
com.neo4j.server.enterprise.EnterpriseEntryPoint
--home-dir=/var/lib/neo4j
--config-dir=/var/lib/neo4j/conf
```

You can see that the OnOutOfMemoryError parameter has been incorrectly split into 4 separate parameters.
Up until Neo4j 4.4.8 this issue was not present.

In case the format in which parameters with spaces have to be provided within `dbms.jvm.additional` changed, please refer me to the documentation.

**Neo4j Version:** 4.4.9
**Operating System:** RHEL 7.9 w/ Docker 20.10.5
**API:** Docker

### Steps to reproduce
Try to start a Neo4j docker container with the parameters provided above:
```
docker run --env=NEO4J_ACCEPT_LICENSE_AGREEMENT=yes --env=NEO4J_dbms_jvm_additional='"-XX:+HeapDumpOnOutOfMemoryError" "-XX:HeapDumpPath=/tmp/" "-XX:OnOutOfMemoryError=/usr/bin/env handle-oom.sh %p /tmp/java_pid%p.hprof" "-Dlog4j.configurationFile=classpath:log4j2.json"' neo4j:4.4.9-enterprise
```

### Expected behavior
Neo4j should start up without errors showing an output like this:
```
2022-08-03 13:01:24.576+0000 INFO  Starting...
2022-08-03 13:01:25.310+0000 INFO  This instance is ServerId{7d2fcb67} (7d2fcb67-eb01-4ad8-b906-e5dad1949c8f)
2022-08-03 13:01:27.847+0000 INFO  ======== Neo4j 4.4.9 ========
2022-08-03 13:01:31.714+0000 INFO  Sending metrics to CSV file at /var/lib/neo4j/metrics
2022-08-03 13:01:31.786+0000 INFO  Bolt enabled on 0.0.0.0:7687.
2022-08-03 13:01:33.369+0000 INFO  Remote interface available at http://localhost:7474/
2022-08-03 13:01:33.377+0000 INFO  id: 873067D4D1C8CD6469576A7FDCAC8B881C3EABFE7919C3FBCB14F2536E6BDD72
2022-08-03 13:01:33.378+0000 INFO  name: system
2022-08-03 13:01:33.378+0000 INFO  creationDate: 2022-08-03T13:01:28.93Z
2022-08-03 13:01:33.379+0000 INFO  Started.
```

### Actual behavior
Neo4j fails to start with the following output:
```
Error: Could not find or load main class "-XX:OnOutOfMemoryError=.usr.bin.env
Caused by: java.lang.ClassNotFoundException: "-XX:OnOutOfMemoryError=.usr.bin.env
```

**Comment by nioertel** [comment]
I also raised issue neo4j/neo4j#12914 but realized that the problem is more likely to be in the entrypoint script.

**Comment by jennyowen** [comment]
Hi, sorry for the delay in addressing this.
Thank you for including reproduction code. It looks as though this issue is caused by https://github.com/neo4j/docker-neo4j/pull/349 which was a bugfix to allow a dollar character to be passed in to jvm.additional.
I've been looking into it, but so far I've not been able to come up with a solution that works for both dollar and space characters. I'm not sure it's possible.

My personal preference would be to revert the old bugfix to make the 4.4.9 behaviour cosistent with the other 4.4 images, but that would be knowingly re-introducing a bug. I'm going to have to consult the customer support team to figure out what's best.

**Comment by nioertel** [comment]
FYI: I have related ticket #20000 open with enterprise customer support.

**Comment by nioertel** [comment]
Just an idea - not sure about side effects, though:
```bash
#!/bin/bash

function params_test()
{
  echo "----------------------------------"
  for param in "$@"; do
    echo "   '$param'"
  done
  echo "----------------------------------"
}

COMMAND='params_test 1 "2 3" 4 x$x'

echo 'Running "$COMMAND" (-> quoting issue)'
$COMMAND
echo

echo 'Running "eval $COMMAND" (-> dollar issue)'
eval $(echo $COMMAND)
echo

echo 'Running improved "eval $COMMAND"'
eval $(echo $COMMAND | sed 's/\$/\\$/g')
echo
```

**Comment by nioertel** [comment]
Having given this some more thoughts, I think that #349 is conceptually wrong.
Running this command:
```bash
docker run --env=NEO4J_ACCEPT_LICENSE_AGREEMENT=yes --env=NEO4J_dbms_jvm_additional="'"'-Dhello=world$x'"' -Dfoo=bar" neo4j:4.4.9-enterprise
```
Leads to this entry in neo4j.conf (-> as expected):
```bash
dbms.jvm.additional='-Dhello=world$x' -Dfoo=bar
```
However `neo4j console --dry-run` prints:
```bash
/opt/java/openjdk/bin/java -cp /var/lib/neo4j/plugins/*:/var/lib/neo4j/conf/*:/var/lib/neo4j/lib/* -Dhello=world$x -Dfoo=bar -Dfile.encoding=UTF-8 com.neo4j.server.enterprise.EnterpriseEntryPoint --home-dir=/var/lib/neo4j --config-dir=/var/lib/neo4j/conf
```
My expectation would be that `neo4j console --dry-run` creates a command that can be c/p executed, such as:
```bash
/opt/java/openjdk/bin/java -cp /var/lib/neo4j/plugins/*:/var/lib/neo4j/conf/*:/var/lib/neo4j/lib/* '-Dhello=world$x' -Dfoo=bar -Dfile.encoding=UTF-8 com.neo4j.server.enterprise.EnterpriseEntryPoint --home-dir=/var/lib/neo4j --config-dir=/var/lib/neo4j/conf
```

So the fix for #349 should be implemented rather in the neo4j command than in the entrypoint.


Niels

**Comment by jennyowen** [comment]
OK this should be fixed in 4.4.11 and 4.3.18.
We basically ended up doing the same thing you suggested, which was have `--dry-run` add correct quoting to arguments. Then also enabling it back with https://github.com/neo4j/docker-neo4j/pull/376

The fixed Neo4js will be in dockerhub once this PR is accepted by the team at Docker:
https://github.com/docker-library/official-images/pull/13102
